tests/pc/test_lag_member.py

```diff
@@ -0,0 +1,356 @@
+    mask_len = vlan["ip"].split("/")[1]
+    lag_ip = "{}.2/{}".format(ip_prefix, mask_len)
+    port_not_behind_lag_ip = "{}.3/{}".format(ip_prefix, mask_len)
+    # Add lag
+    ptfhost.create_lag(PTF_LAG_NAME, lag_ip, "802.3ad")
+
+    port_list = []
+    # Add member to lag
+    for _, port_name in ptf_ports[ATTR_PORT_BEHIND_LAG].items():
+        ptfhost.add_intf_to_lag(PTF_LAG_NAME, port_name)
+        port_list.append(port_name)
+
+    lag_port_map = {}
+    lag_port_map[PTF_LAG_NAME] = {
+        "port_list": port_list,
+        "ip": lag_ip
+    }
+    lag_port_map[ATTR_PORT_NOT_BEHIND_LAG] = ptf_ports[ATTR_PORT_NOT_BEHIND_LAG]
+    lag_port_map[ATTR_PORT_NOT_BEHIND_LAG]["ip"] = port_not_behind_lag_ip
+
+    ptfhost.startup_lag(PTF_LAG_NAME)
+    ptfhost.add_ip_to_dev(ptf_ports[ATTR_PORT_NOT_BEHIND_LAG]["port_name"], port_not_behind_lag_ip)
+    # Wait for lag sync
+    time.sleep(10)
+
+    return lag_port_map
+
+def setup_dut_ptf(ptfhost, duthost, tbinfo):
+    """
+    Setup dut and ptf based on ports available in dut vlan
+
+    Args:
+        ptfhost: PTF host object
+        duthost: DUT host object
+        tbinfo: fixture provides information about testbed
+
+    Returns:
+        information about lag of ptf and dut
+    """
+    # Get number of members in a lag
+    dut_hwsku = duthost.facts["hwsku"]
+    number_of_lag_member = HWSKU_INTF_NUMBERS_DICT.get(dut_hwsku, DEAFULT_NUMBER_OF_MEMBER_IN_LAG)
+
+    # Get id of vlan that concludes enough up ports as src_vlan_id, port in this vlan is used for testing
+    cfg_facts = duthost.config_facts(host = duthost.hostname, source="running")["ansible_facts"]
+    port_status = cfg_facts["PORT"]
+    src_vlan_id = -1
+    pytest_require(cfg_facts.has_key("VLAN_MEMBER"), "Can't get vlan member")
+    for vlan_name, members in cfg_facts["VLAN_MEMBER"].items():
+        # Number of members in vlan is insufficient
+        if len(members) < number_of_lag_member + 1:
+            continue
+
+        # Get count of available port in vlan
+        count = 0
+        for vlan_member in members:
+            if port_status[vlan_member].get("admin_status", "down") != "up":
+                continue
+
+            count += 1
+            if count == number_of_lag_member + 1:
+                src_vlan_id = int(''.join([i for i in vlan_name if i.isdigit()]))
+                break
+
+        if src_vlan_id != -1:
+            break
+
+    pytest_require(src_vlan_id != -1, "Can't get usable vlan concluding enough member")
+
+    dut_ports = {
+        ATTR_PORT_BEHIND_LAG: {},
+        ATTR_PORT_NOT_BEHIND_LAG: {}
+    }
+    src_vlan_members = cfg_facts["VLAN_MEMBER"]["Vlan{}".format(src_vlan_id)]
+    # Get the port correspondence between DUT and PTF
+    port_index_map = cfg_facts["port_index_map"]
+    # Get dut_ports (behind / not behind lag) used for creating dut lag by src_vlan_members and port_index_map
+    for port_name, _ in src_vlan_members.items():
```

Comment:
I realized that there is a problem with this part in last commit: there is no guarantee that the port obtained here must be in the vlan where we will remove the member. So I modified this part to ensure that getting available ports is completely based on the src_vlan obtained from the previous